tests/cupy_tests/functional_tests/test_vectorize.py

```diff
@@ -615,3 +615,19 @@ def my_func(x1, x2, x3):
         x2 = testing.shaped_random((20, 30), xp, dtype, seed=2)
         x3 = testing.shaped_random((20, 30), xp, dtype, seed=3)
         return f(x1, x2, x3)
+
+    @testing.numpy_cupy_array_equal()
+    def test_vectorize_lambda(self, xp):
+        f = xp.vectorize(lambda a, b, c: a + b * c)
+        x1 = testing.shaped_random((20, 30), xp, numpy.int64, seed=1)
+        x2 = testing.shaped_random((20, 30), xp, numpy.int64, seed=2)
+        x3 = testing.shaped_random((20, 30), xp, numpy.int64, seed=3)
+        return f(x1, x2, x3)
+
+    def test_vectorize_lambda_xfail(self):
+        functions = [lambda a, b: a + b, lambda a, b: a * b]
+        f = cupy.vectorize(functions[0])
```

Comment:
Why do we expect an error with a single lambda function?

Reply:
In this testcase, the result of `inspect.getsource` is `functions = [lambda a, b: a + b, lambda a, b: a * b]`, not `lambda a, b: a + b`. So we cannot detect which lambda function is the target.

Reply:
Ahh interesting, thanks for explanation @asi1024!